app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -1247,6 +1250,27 @@ class PluginApi {
     Composer.reopen({ beforeSave: method });
   }
 
+  /**
+   * Registers a callback function to handle the composer save errors.
+   * This allows you to implement custom logic that will happen before
+   * the raw error is presented to the user in bootbox modal.
+   * The passed function is expected to return true if the error was handled,
+   * false otherwise.
+   *
+   * Example:
+   *
+   * api.addComposerSaveErrorCallback((error) => {
+   *   if (error == "my_error") {
+   *      //handle error
+   *      return true;
```

Comment:
I'm a little unsure what `return true/false` means here... does one of them skip displaying the error dialog to the user?

Reply:
Yes, if a single plugin consumer returns true we skip the bootbox, aka the default Discourse error handling in this flow.